### ISSUE TITLE
unused_enumerated rule should catch map/filter/flatMap usages

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2560]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

```swift
// This should trigger a violation:
array.enumerated().map { _, elem in elem }
```
